pyclowder/connectors.py

```diff
@@ -512,18 +514,21 @@ def register_extractor(self, endpoints):
 
         headers = {'Content-Type': 'application/json'}
         data = self.extractor_info
+        if self.extractor_key is not None and len(self.extractor_key) > 0:
```

Comment:
```suggestion
        if self.extractor_key:
```
This will check to make sure it is not None, and the is not empty string.